app/workers/order_processor.rb

```diff
@@ -0,0 +1,43 @@
+class OrderProcessor
+  include Sidekiq::Worker
+  sidekiq_options backtrace: true, queue: :deliveries
+  def perform(consumed_order)
+    begin
+      puts 'sidekiq worker processed ' + consumed_order.to_s
+      consumed_order = JSON.parse(consumed_order)
+      order = Order.find_or_initialize_by(order_id: consumed_order['order_id'] )
+      if order.publish_time.nil? || order.publish_time < consumed_order['publish_time']
```

Comment:
it's a little difficult to understand that order is either created or updated
then in the rest of the flow you check whether order is new by checking that its values are nil
I think it should be separated to 2 different methods - one that creates and one that updates

Reply:
i'm always upserting so it doesn't matter new Order or not .. the nil check are for merging purposes as i don't want nil overrides.. same order can arrive multiple times with added attributes and i want to merge them always